src/Parser.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace CSVParser;
+
+class Parser {
+
+    var $path;

```

Comment:
Since I specified this project requires PHP7.1 in `Composer.json` you will need to specify the visibility of these member variables otherwise it will cause a syntax error. I would also recommend adding some PHPDoc comments so any developer that comes behind you know how those variables are supposed to represent. Unless you can provide sufficient reason to use pubic or protected as the visibility modifiers I am firmly requesting you specify them as private variables. Also note I renamed `headers_array` to just `headers` This was purely a style change because I don't think the _array added anything meaningful.

i.e.
```php
/*
 * @var $path The file path to the CSV file to be parsed.
 */
private $path;

/*
 * @var $headers The column headers for the CSV file that give context to the column values
 */
$headers;

/*
 * @var $delimiter The delimiter value that separates column values for this CSV File.
 */
$delimiter;
```

Reply:
I do have many comments, just not written in the way you have.  I used the // in order comment lines.  I do suspect they could use more information on a few of them through.

With no declaration in front of headers and delimiter does it auto assume private?

Reply:
You do have plenty of comments. What I have suggested is just a standardized format that IDE's can integrate into intellisense. Other PHP developers will also be familiar with it.